feature/navigation/drawer/src/main/kotlin/app/k9mail/feature/navigation/drawer/ui/folder/FolderListItem.kt

```diff
@@ -37,8 +45,35 @@ fun FolderListItem(
     )
 }
 
-private fun mapFolderIcon(type: FolderType): ImageVector {
-    return when (type) {
+@Composable
+private fun mapFolderName(displayFolder: DisplayFolder): String {
+    val resources = LocalContext.current.resources
+    val folderNameFormatter = remember { FolderNameFormatter(resources) }
```

Comment:
This will create one `FolderNameFormatter` per list item. 

We should probably create an instance at the drawer level and pass the instance to the list item composables.

Reply:
Yes, I'll move it up